gradle/libs.versions.toml

```diff
@@ -0,0 +1,153 @@
+# Also see settings.gradle file for dynamic versionCatalogs dependencies
+
+[versions]
+antlr = "4.9.3"
+deltaStandalone = "0.6.0"
+deltaCore = "2.2.0"
+flink115 = { strictly = "[1.15, 1.16[", prefer = "1.15.0"}
+flink116 = { strictly = "[1.16, 1.17[", prefer = "1.16.2"}
+flink117 = { strictly = "[1.17, 1.18[", prefer = "1.17.1"}
+guava = "32.1.1-jre"
+hadoop2 = "2.7.3"
+hive2 = { strictly = "[2, 3[", prefer = "2.3.9"}
+hive3 = "3.1.3"
+jackson211 = "2.11.4"
+jackson212 = "2.12.3"
+jackson213 = "2.13.4"
+jetty = "9.4.43.v20210629"
+junitJupiter = "5.9.2"
+mockito = "4.11.0"
+nessie = "0.63.0"
+orc = "1.9.0"
+parquet = "1.13.1"
+scalaCollectionCompat = "2.6.0"
+sparkHive31 = "3.1.3"
+sparkHive32 = "3.2.2"
+sparkHive33 = "3.3.2"
+sparkHive34 = "3.4.1"
+springBoot = "2.5.4"
+tez010 = "0.10.2"
+tez08 = { strictly = "[0.8, 0.9[", prefer = "0.8.4"}
```

Comment:
any particular reason for the version range vs just defining the version?

Reply:
For places like this(where a rich version declaration with  strictly & prefer is used) and similar others I will put a single comments: please read a [head description to PR](https://github.com/apache/iceberg/pull/7694#issue-1723184954), I've updated it with a point about rich versions. Let me know if you have questions and/or suggestions.